src/domain/session/room/ComposerViewModel.js

```diff
@@ -0,0 +1,81 @@
+import {ViewModel} from "../../ViewModel.js";
+
+export class ComposerViewModel extends ViewModel {
+    constructor(roomVM) {
+        super();
+        this._roomVM = roomVM;
+        this._isEmpty = true;
+        this._replyId = null;
+        this._replyVM = null;
+    }
+
+    setReplyingTo(entry) {
+        const newId = entry?.id || null;
+        const changed = this._replyId !== newId;
+        if (changed) {
+            this._replyId = newId;
+            if (this._replyVM) {
+                this.untrack(this._replyVM);
+                this._replyVM.dispose();
+            }
+            this._replyVM = entry && this._roomVM._createTile(entry);
+            if (this._replyVM) {
+                this.track(this._replyVM);
+            }
+            this.emitChange("replyViewModel");
+        }
+    }
+
+    clearReplyingTo() {
+        this.setReplyingTo(null);
+    }
+
+    get replyViewModel() {
+        return this._replyVM;
+    }
+
+    get isEncrypted() {
+        return this._roomVM.isEncrypted;
+    }
+
+    sendMessage(message) {
+        const success = this._roomVM._sendMessage(message, this._replyVM);
```

Comment:
oops, this was already missing an await before your PR. Can you add it please? :pray:

Reply:
Does this mean making `sendMessage` and `_trySend` async, as well? They depend on the results of `_sendMessage`.